### DISCUSSION TITLE
Get Date with Range Date in a columns

### DISCUSSION THREAD

**Discussion by chrissolontio** [teamalgoritma/community#141] [Q&A] [answered]
Dear All,

Ketika Saya run kode dibawah ini ada beberapa tanggal di kolom Tglsjp tidak keambil. ada yang salah kah? atau ada alternatif lainnya
`all_data[(all_data["Tglsjp"] > '2021-01-31') & (all_data["Tglsjp"] <= '2022-01-31')]`

*Note : Data Type "Tglsjp" is datetime64[ns]

**Comment by onlyphantom** [comment, marked as answer]
For `datetime64`, you will want to use `pd.Timestamp()` for proper filtering:

```
all_data[(all_data["Tglsjp"] > pd.Timestamp(2021, 1, 31)) & (all_data["Tglsjp"] <= pd.Timestamp(2022, 1, 31)]
```

An alternative is to bring in `strftime` to convert your datetime to dates:

```
all_data[all_data['Tglsjp'].dt.strftime('%Y-%m-%d') > '2021-01-31']
```
